pykechain/models/customization.py

```diff
@@ -167,6 +167,111 @@ def add_json_widget(self, config):
         validate(config, component_json_schema)
         self._add_widget(dict(config=config, name="jsonWidget"))
 
+    def add_super_grid_widget(self, part_model, delete=False, edit=True, export=True, incomplete_rows=True,
+                              new_instance=False, parent_part_instance=None, max_height=None, custom_title=None,
+                              emphasize_edit=False, emphasize_new_instance=True):
+        """
+        Add an KE-chain superGrid (e.g. basic table) widget to the customization.
+
+        The widget will be saved to KE-chain.
+
+        :param emphasize_new_instance:
+        :type emphasize_new_instance: bool
+        :param emphasize_edit:
+        :type emphasize_edit: bool
+        :param new_instance:
+        :type new_instance: bool
+        :param incomplete_rows:
+        :type incomplete_rows: bool
+        :param export:
+        :type export: bool
+        :param edit:
+        :type edit: bool
+        :param delete:
+        :type delete: bool
+        :param part_model: The part model based on which all instances will be shown.
+        :type parent_part_instance: :class:`Part`
+        :param parent_part_instance: The parent part instance for which the instances will be shown or to which new
+        instances will be added.
+        :type parent_part_instance: :class:`Part`
+        :param max_height: The max height of the supergrid in pixels
+        :type max_height: int or None
+        :param custom_title: A custom title for the supergrid
+                 - None (default): Part instance name
+                 - String value: Custom title
+                 - False: No title
+        :type custom_title: basestring or None
+        """
+        # Assertions
+        if not (new_instance and bool(parent_part_instance)):
+            raise IllegalArgumentError("If you want to allow the creation of new part instances, you must specify a "
+                                       "parent_part_instance")
+
+        # Declare supergrid config
+        config = {
+            "xtype": ComponentXType.SUPERGRID,
+            "filter": {
+                "activity_id": str(self.activity.id),
+                "model": str(part_model.id)
+            },
+            "viewModel": {
+                "data": {
+                    "actions": {
+                        "delete": delete,
+                        "edit": edit,
+                        "export": export,
+                        "incompleteRows": incomplete_rows,
+                        "newInstance": new_instance
+                    },
+                    "ui": {
+                        "newInstance": "primary-action" if emphasize_new_instance else "default-toolbar",
```

Comment:
are you sure this is correct? I don't know this, so can be totally correct.
I would normally not introduce if clauses like this. I would rewrite this as follows
```
"newInstance": emphasize_new_instance and "primary-action" or "default-toolbar",
```